components/raftstore/src/store/fsm/peer.rs

```diff
@@ -154,6 +152,14 @@ where
     /// which is reporting the store/peer meta to PD.
     unsafe_recovery_target_commit_index: Option<u64>,
     unsafe_recovery_wait_apply_counter: Option<Arc<AtomicUsize>>,
+    /// Total number of recovery tasks to be finished. Every missioned peer subtract 1 from this
+    /// number after its own task is finished, the last peer is also responsible for reporting to
+    /// PD.
+    unsafe_recovery_num_tasks: Option<Arc<AtomicUsize>>,
```

Comment:
can we introduce a `UnsafeRecoveryState`?

Reply:
you can make it to a state machine, then the logic would be clearer without these bool fields